config/initializers/seeds.rb

```diff
@@ -0,0 +1,22 @@
+return unless (DatabaseSeed.table_exists? rescue(false)) && !Foreman.in_rake?('db:rollback') && !Foreman.in_rake?('db:seed') && !Foreman.in_rake?('db:migrate')
+
+Foreman::Application.configure do |app|
+  config.after_initialize do
+    seeder = ForemanSeeder.new
+
+    db_seed = DatabaseSeed.find_by_id(1)
+    db_seed = DatabaseSeed.create if db_seed.nil?
+
+    old_hash = db_seed.file_hash
+    new_hash = seeder.hash
+
+    if new_hash != old_hash
+      seeder.execute
+
+      db_seed.file_hash = new_hash
+      db_seed.save!
```

Comment:
I think this is still valid - the hash should be updated if we run the seeds manually as well

Reply:
I was resisting this because its a minor savings but adds complexity to the ForemanSeeder class as it would now be tied to knowledge of database entities. I can go that route still just wanted to clarify my reasoning.